dist/tools/ethos/ethos.c

```diff
@@ -605,7 +608,21 @@ int main(int argc, char *argv[])
 
         if (FD_ISSET(STDIN_FILENO, &readfds)) {
             ssize_t res = read(STDIN_FILENO, inbuf, sizeof(inbuf));
-            if (res <= 0) {
+            if (res == 0) {
+                fprintf(stderr, "EOF from stdin\n");
+                if (isatty(STDIN_FILENO)) {
```

Comment:
can the isatty call be moved outside the while loop? It shouldn't change between iterations, right?

Reply:
Yes, it should be constant, but I thought it looked better like this. It will only be called whenever an EOF is sent from the terminal.

Reply:
The isatty call will only be made once, because of the `if (stdin_open)` above, and the `break` below.

Reply:
I was thinking the same thing before, why not close it right away. But doing it there allows doing `{ sleep 5; echo ifconfig; sleep 1; } | make term`